packages/flutter_tools/lib/src/build_runner/build_script.dart

```diff
@@ -0,0 +1,459 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: implementation_imports
+import 'dart:async';
+import 'dart:convert'; // ignore: dart_convert_import
+import 'dart:io'; // ignore: dart_io_import
+import 'dart:isolate';
+
+import 'package:analyzer/analyzer.dart'; // ignore: deprecated_member_use
+import 'package:build_runner/build_runner.dart' as build_runner;
+import 'package:build/build.dart';
+import 'package:build_config/build_config.dart';
+import 'package:build_modules/build_modules.dart';
+import 'package:build_modules/builders.dart';
+import 'package:build_modules/src/module_builder.dart';
+import 'package:build_modules/src/platform.dart';
+import 'package:build_modules/src/workers.dart';
+import 'package:build_runner_core/build_runner_core.dart' as core;
+import 'package:build_test/builder.dart';
+import 'package:build_test/src/debug_test_builder.dart';
+import 'package:build_web_compilers/build_web_compilers.dart';
+import 'package:build_web_compilers/builders.dart';
+import 'package:build_web_compilers/src/dev_compiler_bootstrap.dart';
+import 'package:crypto/crypto.dart';
+
+import 'package:path/path.dart' as path; // ignore: package_path_import
+import 'package:scratch_space/scratch_space.dart';
+import 'package:test_core/backend.dart';
+
+const String ddcBootstrapExtension = '.dart.bootstrap.js';
+const String jsEntrypointExtension = '.dart.js';
+const String jsEntrypointSourceMapExtension = '.dart.js.map';
+const String jsEntrypointArchiveExtension = '.dart.js.tar.gz';
+const String digestsEntrypointExtension = '.digests';
+const String jsModuleErrorsExtension = '.ddc.js.errors';
+const String jsModuleExtension = '.ddc.js';
+const String jsSourceMapExtension = '.ddc.js.map';
+
+final DartPlatform flutterWebPlatform =
+    DartPlatform.register('flutter_web', <String>[
+  'async',
+  'collection',
+  'convert',
+  'core',
+  'developer',
+  'html',
+  'html_common',
+  'indexed_db',
+  'js',
+  'js_util',
+  'math',
+  'svg',
+  'typed_data',
+  'web_audio',
+  'web_gl',
+  'web_sql',
+  '_internal',
+  // Flutter web specific libraries.
+  'ui',
+  '_engine',
+  'io',
+  'isolate',
+]);
+
+/// The build application to compile a flutter application to the web.
```

Comment:
nit: there seem to be grammar issues in this sentence, because I do not understand what it says :thinking: Is the first "application" a nounization of the `core.apply()` call? Maybe call it "execution"?

Reply:
rephrased to be less grammatically dubious